test/test_namedtensor.py

```diff
@@ -1054,6 +1062,11 @@ def test_simple_reduce(op, device):
             with self.assertRaisesRegex(RuntimeError, 'Name \'H\' not found'):
                 op(t, 'H')
 
+        def test_autograd_supports_dimname_overload(op, device):
+            t = torch.empty(2, 3, 5, names=('N', 'C', 'L'), device=device, requires_grad=True)
+            sum_all_outputs(op(t, 'C')).backward()
+            self.assertTrue(t.grad is not None)
```

Comment:
can we do anything to validate the value of the gradient? would it be helpful to validate the grad against the equivalent op on an unnamed tensor?

Reply:
nit: btw I think there's a self.assertIsNotNone()

Reply:
> can we do anything to validate the value of the gradient?

We can, but see the next point

> would it be helpful to validate the grad against the equivalent op on an unnamed tensor?

Because autograd ignores all names on tensors, we don't need to validate the grad as long as we test that "autograd ignores all names on tensors"

> nit: btw I think there's a self.assertIsNotNone()

Sure I'll use that

Reply:
>autograd ignores all names on tensors 

will that ever not be the case for any kinds of gradients?

I think you had also mentioned an issue with the correctness of the mean gradient that we were also fixing, though maybe that's (going to be) a separate PR?

Reply:
> > autograd ignores all names on tensors
> 
> will that ever not be the case for any kinds of gradients?

In the future we may design and implement name-aware autograd. This is not going to happen soon. When that happens, we can revisit the testing plan for this.

> I think you had also mentioned an issue with the correctness of the mean gradient that we were also fixing, though maybe that's (going to be) a separate PR?

yeah, that's orthogonal to this PR. It is this one: https://github.com/pytorch/pytorch/issues/29137#issuecomment-549938081

Reply:
This PR fixes https://github.com/pytorch/pytorch/issues/28286

Reply:
thanks for the clarification